ui/shared/AssetAndAmountInput.qml

```diff
@@ -0,0 +1,134 @@
+  anchors.left: parent.left
+
+
+  Item {
+    id: txtBalanceInfo
+    anchors.right: parent.right
+    anchors.left: parent.left
+    anchors.top: parent.top
+    anchors.topMargin: 0
+    height: txtBalanceDesc.height
+
+    StyledText {
+        id: txtBalanceDesc
+        text: qsTr("Balance: ")
+        anchors.right: txtBalance.left
+        font.weight: Font.Medium
+        font.pixelSize: 13
+        color: Style.current.grey
+    }
+
+    StyledText {
+        id: txtBalance
+        text: selectAsset.selectedAsset.value
+        anchors.right: parent.right
+        font.weight: Font.Medium
+        font.pixelSize: 13
+        color: Style.current.grey
+
+        MouseArea {
+            cursorShape: Qt.PointingHandCursor
+            anchors.fill: parent
+            hoverEnabled: true
+            onExited: {
+                txtBalance.color = Style.current.grey
+            }
+            onEntered: {
+                txtBalance.color = Style.current.black
+            }
+            onClicked: {
+                inputAmount.setModel(selectAsset.selectedAsset.value)
+            }
+        }
+    }
+  }
+
+  Input {
+      id: inputAmount
+      label: qsTr("Asset & Amount")
+      placeholderText: "0.00"
+      validationError: root.errorMessage
+      anchors.top: parent.top
+      customHeight: 56
+      onTextChanged: {
+        root.modelChanged(inputAmount.text)
+      }
+      onModelChanged: {
+        let amount = inputAmount.text.trim()
+        if (amount === "" || isNaN(amount)) {
+          return
+        }
+        txtFiatBalance.setModel(walletModel.getFiatValue(amount, selectAsset.selectedAsset.symbol, "USD"))
+      }
+  }
+
+  AssetSelector {
+      id: selectAsset
+      assets: root.assets
+      width: 86
+      height: 28
+      anchors.top: inputAmount.top
+      anchors.topMargin: Style.current.bigPadding + 14
+      anchors.right: parent.right
+      anchors.rightMargin: Style.current.smallPadding
+      onAssetSelected: function (selectedAsset) {
+        inputAmount.setModel(selectedAsset.value)
+      }
+  }
+
+  Item {
+    height: txtFiatBalance.height
+    anchors.left: parent.left
+    anchors.top: inputAmount.bottom
+    anchors.topMargin: inputAmount.labelMargin
+
+    StyledTextField {
+      id: txtFiatBalance
+      anchors.left: parent.left
+      anchors.top: parent.top
+      color: txtFiatBalance.activeFocus ? Style.current.black : Style.current.grey
+      font.weight: Font.Medium
+      font.pixelSize: 12
+      inputMethodHints: Qt.ImhFormattedNumbersOnly
+      text: "0.00"
+      selectByMouse: true
+      background: Rectangle {
+        color: Style.current.transparent
+      }
+      padding: 0
+      onModelChanged: {
+        let balance = txtFiatBalance.text.trim()
+        if (balance === "" || isNaN(balance)) {
+          return
+        }
+        inputAmount.setModel(walletModel.getCryptoValue(balance, "USD", selectAsset.selectedAsset.symbol))
+      }
```

Comment:
Same here `onModelChanged` was introduced in `StyledTextField` to account for infinite loops